packages/engine/lib/execution/src/package/simulation/output/parts.rs

```diff
@@ -107,27 +104,21 @@ impl OutputPartBuffer {
         Ok(())
     }
 
-    pub fn finalize(mut self) -> Result<(Vec<u8>, Vec<PathBuf>)> {
+    pub fn finalize(&mut self) -> Result<&[PathBuf]> {
         self.current.push(CHAR_OPEN_RIGHT_SQUARE_BRACKET);
         self.persist_current_on_disk()?;
-        Ok((self.current, self.parts))
+        Ok(&self.parts)
     }
+}
 
-    pub fn remove_experiment_parts(experiment_id: ExperimentId) {
-        let path = format!("{RELATIVE_PARTS_FOLDER}/{experiment_id}");
-        match std::fs::remove_dir_all(&path) {
-            Ok(_) => {
-                tracing::trace!(
-                    experiment = %experiment_id,
-                    "Removed parts folder for experiment {experiment_id}: {path:?}"
-                );
-            }
-            Err(err) => {
-                tracing::warn!(
-                    experiment = %experiment_id,
-                    "Could not clean up {path:?}: {err}"
-                );
-            }
+impl Drop for OutputPartBuffer {
+    fn drop(&mut self) {
+        for part in &self.parts {
+            std::fs::remove_file(part)
+                .unwrap_or_else(|error| tracing::error!("Failed to remove part {part:?}: {error}"))
         }
+        std::fs::remove_dir_all(&self.base_path).unwrap_or_else(|error| {
```

Comment:
This looks like the same logic as before, so this question might be about if our old mechanism was broken, but:

Does this happily error and move along if another experiment is using the folder? Slightly concerned about a race condition if another experiment just passed the point in code where it creates a part folder, and then this goes and deletes it.

Reply:
Previously, `base_path` was hard-coded, but now it's a temporary directory. However, I improved the error message a bit: 1d7213bcdf1e223fbec94e31a9aec603baf20204

Reply:
A `base_path` is created for each `start_new_sim_run`